jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/ExtraDirectoriesParameters.java

```diff
@@ -98,6 +101,22 @@ public void setPaths(Object paths) {
             .collect(Collectors.toList()));
   }
 
+  /**
+   * Sets paths, for lazy evaluation where {@code paths} is a {@link Provider} of a suitable object.
+   *
+   * @param paths provider of paths to set
+   * @see #setPaths(Object)
+   */
+  public void setPaths(Provider<Object> paths) {
```

Comment:
Ah I could be oversimplifying this in my mind but the Provider API should allow us to avoid setters all together. Similar to what is being done in https://github.com/GoogleContainerTools/jib/pull/3242, would it be possible to just modify the getter to return a `Property` and remove the setters?

Reply:
Yeah I was pondering about this as well! 

With eliminating `setPermissions` this was able to work pretty straightforwardly, but here (in `setPaths` and `setFrom`) the existing logic to convert object to file paths is more complicated than just value assignment, and I couldn’t find a way for the Gradle-generated setters to handle it. 

With `setInto`, it also looked like a larger change to have the getters return `Property` since it would also change up the shared `ExtraDirectoriesConfiguration` interface. Not sure what the best approach here would be?

Reply:
Makes sense, the consideration was definitely backward compatibility here. Will keep the guidelines in mind for new parameters in the future.